creator-node/test/tracks.js

```diff
@@ -445,45 +447,39 @@ describe('test /track_content with actual ipfsClient', function () {
     let storagePath = config.get('storagePath')
     storagePath = storagePath.slice(0, 1) === '/' ? '.' + storagePath : storagePath
 
-    // check if track UUID dir exists
-    const originalTrackUUID = resp.body.source_file.split('.').slice(0, -1).join('.') // remove extension
-    const originalTrackUUIDPath = path.join(storagePath, originalTrackUUID)
-    assert.ok(fs.existsSync(originalTrackUUIDPath))
-
-    // check that the track UUID dir contains the transcoded copy
-    const transcodedTrackPath = path.join(originalTrackUUIDPath, originalTrackUUID + '-dl.mp3')
-    assert.ok(fs.existsSync(transcodedTrackPath))
-
     // check that the generated transcoded track is the same as the transcoded track in /tests
     const transcodedTrackAssetPath = path.join(__dirname, 'testTranscoded320Track.mp3')
     const transcodedTrackAssetBuf = fs.readFileSync(transcodedTrackAssetPath)
+    const transcodedTrackPath = path.join(storagePath, resp.body.data.transcodedTrackCID)
     const transcodedTrackTestBuf = fs.readFileSync(transcodedTrackPath)
     assert.deepStrictEqual(transcodedTrackAssetBuf.compare(transcodedTrackTestBuf), 0)
 
-    // check that the track UUID dir contains the source file
-    const sourceFile = resp.body.source_file
-    const sourceFilePath = path.join(originalTrackUUIDPath, sourceFile)
-    assert.ok(fs.existsSync(sourceFilePath))
-
-    // check that there are 32 segments in <uuid>/segments and that they follow
-    // the naming convention 'segment<3 digit #>.ts'
-    const segmentsPath = path.join(originalTrackUUIDPath, 'segments')
-    fs.readdir(segmentsPath, (err, files) => {
-      if (err) assert.fail(err.message)
-      assert.deepStrictEqual(files.length, 32)
-
-      for (let i = 0; i < 32; i++) {
-        const indexSuffix = ('000' + i).slice(-3)
-        assert.deepStrictEqual(files[i], `segment${indexSuffix}.ts`)
-      }
-    })
-
-    // check that there are 32 CIDs that have been added to fs
+    // Ensure 32 segments are returned, each segment has a corresponding file on disk,
+    //    and each segment disk file is exactly as expected
     const segmentCIDs = resp.body.track_segments
-    assert.deepStrictEqual(segmentCIDs.length, 32)
-    segmentCIDs.map(cid => {
+    assert.deepStrictEqual(segmentCIDs.length, testAudiusFileNumSegments)
+    segmentCIDs.map(function (cid, index) {
       const cidPath = path.join(storagePath, cid.multihash)
+
+      // Ensure file exists
       assert.ok(fs.existsSync(cidPath))
+
+      // Ensure file is identical to expected segment file
+      const expectedSegmentFilePath = _getTestSegmentFilePathAtIndex(index)
```

Comment:
here we're assuming that the original track artifacts haven't been pruned?

Reply:
no
this test makes no assumption on the state of artifacts on disk
the track_content route responds with an array of segment CIDs, for each of which there is a file stored in disk at /file_storage/CID
this check compares each of those files to the pre-defined reference files in /test/test-segments
does that make sense? sorry if confusing, can clarify on call

i'm now realizing that these reference segment files are actually stored twice, prob should clean that up but would need to re-point all test references from `/test/test-segments/segment***.ts` to `/test/testTrackUploadDir/testTrackSourceFileDir/segments/segment***.ts` which i can do

Reply:
discussed offline, all clear here

added TODOs to remove duplicative reference segment files